### ISSUE TITLE
Reverse-portrait rotation bug

### ISSUE THREAD

**Issue by mingyee2** [futo-org/grayjay-android#1616]
### What happened?

When turning on system rotation with the “Allow reverse portrait” option off, I expect that, when I rotate my device 180 degrees slowly starting from a portrait orientation, the app orientation should enter landscape and then stay there. Instead, it enters landscape, and then re-enters portrait (but not reverse portrait) as I go to 90 rotation and then to 180 degree rotation. Notably, this only is observed if there is no currently playing video (so even with an open but paused video, this behaviour occurs).

### Grayjay Version

276

### What plugins are you seeing the problem on?

All

### When do you experience the issue?

- [ ] While logged in
- [ ] While logged out
- [X] N/A

**Comment by mingyee2** [comment]
Actually, it appears to happen sometimes even when a video is active, but I have not been able to reliably reproduce it yet.

**Comment by codebymatter** [comment]
Looks like reverse landacape works again. Thanks so much to FUTO for fix! 

Looks like reverse-portrait rotation for app is not working as of recent update. It's not a feature Ivwould use anyway, but for those that do it would affect their usability.

**Comment by Zvonimir-FUTO** [comment]
Hey @codebymatter thank you for testing and reporting back. Reverse-portrait rotation is supposed to be used with system rotation so that's not a bug.

**Comment by codebymatter** [comment]
> Hey @codebymatter thank you for testing and reporting back. Reverse-portrait rotation is supposed to be used with system rotation so that's not a bug.

Ah ok. Thank you for the explaination. Trsted it bc Ibwas curious about it.

**Comment by mingyee2** [comment]
@Zvonimir-FUTO I just saw this was marked as fixed, but it has never appeared fixed on my end. Note that I observe this bug specifically *when a video is not playing*. I still see this on the most recent version (281).

**Comment by Zvonimir-FUTO** [comment]
Couldn't reproduce this. Could you tell me on which android version are you and which device are you using?

**Comment by mingyee2** [comment]
@Zvonimir-FUTO Android 13 on Samsung A12

**Comment by kaidelorenzo** [comment]
@mingyee2 can you explain again what you experience and what your settings are?

**Comment by mingyee2** [comment]
@kaidelorenzo Yes, sure: turning my phone from sideways to upside down changes the app from landscape to portrait mode, but *only if a video is not currently playing*. I have reverse portrait disallowed, and of course screen orientation unlocked. This happens on my A12, but not on an S10 Lite I tested on.

Interestingly, I noticed similar behaviour in the Google Calendar app.

**Comment by mingyee2** [comment]
@kaidelorenzo No, you misunderstand: this *only* happens when a video is not playing. It has the desired behaviour when a video is playing. That makes no sense.

**Comment by kaidelorenzo** [comment]
And it goes to portrait or reverse portrait? Like when you hold you phone upside down it displays grayjay upside down also?

**Comment by mingyee2** [comment]
To be very clear: When the top of the phone (where you would put your ear during a phone call) is oriented lower than the bottom of the phone (so upside-down), the notification bar on my phone is at the top of the phone (so portrait, I guess). It is upside-down with respect to me, if I am right side up, but it is not upside-down with respect to the phone.

Put another way: If I flip the phone 180 degrees, the screen has no change.

**Comment by mingyee2** [comment]
That is correct.
